dev/tests/js/framework/JsTestRunner.php

```diff
@@ -0,0 +1,537 @@
+<?php
+
+class JsTestRunner
```

Comment:
Of course it s not mandatory, using global scope and functions is great, but it just felt like a way to bind the methods to the config data set without using global variables or always passing the array as an argument.

I find smaller methods generally easier to understand and refactor.
To hide the larger number of methods, the class only exposes two public methods, which makes it easier to use, and also document the purpose.

Also, I find it easier to refactor code if I can see which methods are private, that is, implementation details, so they are safe to change.

I expect most of the command line scripts in Magento to be refactored sooner or later to use a common entry point (simply because that is what most PHP devs know from various frameworks).
Having the code in a class might make it easier to reuse some of it in that case.

I've also written some hacky tests (not commited) which compared the output of the class to the output of the originial script, and that also was easier to do with this version of the code in a class.